package Exceptionhandling;

import java.util.Scanner;

public class FinallyRunner {

	public static void main(String[] args) {
		Scanner scanner = null;
		try {
			scanner = new Scanner(System.in);

			int[] numbers = { 1, 2, 3, 4 };

			int number = numbers[5];

			System.out.println("Before Scanner Close");

			// if not called there might be resource or memory leakage

		} catch (Exception ex) {
			ex.printStackTrace();
		} finally {
			scanner.close();
		}

		System.out.println("End");
	}

}
